### PR TITLE
[WIP] - Take Volume Snapshot Action

### PR DESCRIPTION
**Pull request by utchoang**
@rhtyd cc @svenvogel
This is my new design of the Take Volume Snapshot Action. Please review and let me know your comments.
Screenshot_6

### CODE REVIEW

**Comment by rhtyd** [comment]
@utchoang the problem of conflict is there again. Can you check/fix it?

Here's what you can do for future PRs:
- First checkout the latest master and enforce that your version of master is same as origin/master, you can do like: git fetch origin/master && git checkout master && git reset --hard origin/master
- Then for every new issue/task, checkout a feature or bugfix branch after doing above: git checkout -b <name of your branch>

**Comment by utchoang** [comment]
@rhtyd I resolved the conflict PR. Please review again. Thank you

File: src/utils/methods.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import { api } from '@/api'
+import { message, notification } from 'ant-design-vue'
```
Comment (rhtyd) [inline src/utils/methods.js]: @utchoang can you remove this file?

File: src/views/AutogenView.vue [hunk 2]
```diff
@@ -32,7 +32,26 @@
...
+                v-if="action.api in $store.getters.apis &&
+                  ((!dataView && (action.listView || action.groupAction && selectedRowKeys.length > 0)) ||
+                  (dataView && action.dataView && ('show' in action ? action.show(resource) : true)) ||
+                  (treeView && ('show' in action ? action.show(treeSelected) : true)))"
+                :icon="action.icon"
+                :type="action.icon === 'delete' ? 'danger' : (action.icon === 'plus' ? 'primary' : 'default')"
+                shape="circle"
+                style="margin-right: 5px"
+                @click="execAction(action)"
+              >
```
Comment (rhtyd) [inline src/views/AutogenView.vue]: Remove these changes

File: src/views/storage/TakeSnapshot.vue [hunk 3]
```diff
@@ -0,0 +1,264 @@
+<template>
```
Comment (rhtyd) [inline src/views/storage/TakeSnapshot.vue]: Add license

File: src/views/storage/TakeSnapshot.vue [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+  <div class="tak-snapshot">
```
Comment (rhtyd) [inline src/views/storage/TakeSnapshot.vue]: Fix typo `take-snapshot`

File: src/views/storage/TakeSnapshot.vue [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+}
+</script>
+
+<style lang="less" scoped>
+.form {
+  margin-top: 10px;
+}
+
+.tak-snapshot {
+  min-width: 500px;
```
Comment (rhtyd) [inline src/views/storage/TakeSnapshot.vue]: Use this:
```
-.tak-snapshot {
-  min-width: 500px;
+.take-snapshot {
+  width: 85vw;
+
+  @media (min-width: 760px) {
+    width: 500px;
+  }

```

File: src/views/storage/TakeSnapshot.vue [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+          for (const res in json[obj]) {
+            if (res === 'jobid') {
+              hasJobId = true
+              const jobId = json[obj][res]
+              this.$store.dispatch('AddAsyncJob', {
+                title: title,
+                jobid: jobId,
+                description: description,
+                status: 'progress'
+              })
```
Comment (rhtyd) [inline src/views/storage/TakeSnapshot.vue]: Add break here.

File: src/views/storage/TakeSnapshot.vue [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+          formattedTagData['tags[' + i + '].value'] = tag.value
+          params = Object.assign({}, params, formattedTagData)
+        }
+
+        this.actionLoading = true
+        const title = this.$t('label.action.take.snapshot')
+        const description = this.$t('volume') + ' ' + this.resource.id
+        const loading = this.$message.loading(title + 'in progress for ' + description, 0)
+        api('createSnapshot', params).then(json => {
+          this.checkForAddAsyncJob(json, title, description)
```
Comment (rhtyd) [inline src/views/storage/TakeSnapshot.vue]: Could you please use the this.$pollJob() instead (git grep to see other examples)

**Pull request merged**